src/test/HST/ApplicationTests.hs

```diff
@@ -46,6 +46,30 @@ shouldTransformTo input expectedOutput = do
   expectedOutputModule <- parseTestModule expectedOutput
   outputModule `prettyModuleShouldBe` expectedOutputModule
 
+-- | Parses the given modules, initializes the environment for the the input
+--   module at the head of the given list, processes it with 'processModule'
+--   and sets the expectation that the given output module is produced.
+shouldTransformModulesTo
+  :: ( S.EqAST f
+     , Members '[GetOpt, Cancel, Report, SetExpectation, WithFrontend f] r
+     )
+  => [[String]]
+  -> [String]
+  -> Sem r ()
+shouldTransformModulesTo inputs expectedOutput = do
+  inputModules <- mapM parseTestModule inputs
+  let n               = length inputs
+      fileNames       = zipWith (\s n' -> "<" ++ s ++ show n' ++ ">")
+        (replicate n "test-input") [1 .. n]
+      inputModuleList = zip fileNames
+        (map (\m -> (m, createModuleInterface m)) inputModules)
+  env <- runInputModule inputModuleList (initializeEnvironment (head fileNames))
+  outputModule <- runWithEnv env
+    . runFresh (findIdentifiers (head inputModules))
+    $ processModule (head inputModules)
+  expectedOutputModule <- parseTestModule expectedOutput
+  outputModule `prettyModuleShouldBe` expectedOutputModule
+
```

Comment:
There's a lot of code duplication to the `shouldTransformTo` function here. You should either rename the new method to `shouldTransformTo` or change the definition of `shouldTransformTo` to:
`shouldTransformTo input = shouldTransformModulesTo [input]`